templates/eks/amazon-eks-nodepool-cf.yaml

```diff
@@ -2,16 +2,92 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Description: 'Amazon EKS - Node Group'
 
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      -
+        Label:
+          default: "EKS Cluster"
+        Parameters:
+          - ClusterName
+          - ClusterControlPlaneSecurityGroup
+          - NodeSecurityGroup
+      -
+        Label:
+          default: "Worker Node Configuration"
+        Parameters:
+          - NodeGroupName
+          - NodeAutoScalingGroupMinSize
+          - NodeAutoScalingGroupMaxSize
+          - NodeAutoScalingInitSize
+          - NodeVolumeSize
+          - NodeSpotPrice
+          - NodeInstanceType
+          - NodeImageId
+          - KeyName
+          - BootstrapArguments
+      -
+        Label:
+          default: "Worker Network Configuration"
+        Parameters:
+          - VpcId
+          - Subnets
+
 Parameters:
 
+  BootstrapArguments:
+    Description: Arguments to pass to the bootstrap script. See files/bootstrap.sh in https://github.com/awslabs/amazon-eks-ami
+    Default: ""
+    Type: String
+
+  ClusterAutoscalerEnabled:
+      Description: Enable Cluster Autoscaler (true/false)
+      Type: String
+
+  ClusterControlPlaneSecurityGroup:
+      Description: The security group of the cluster control plane.
+      Type: AWS::EC2::SecurityGroup::Id
+
+  ClusterName:
+      Description: The cluster name provided when the cluster was created.  If it is incorrect, nodes will not be able to join the cluster.
+      Type: String
+
   KeyName:
     Description: The EC2 Key Pair to allow SSH access to the instances
     Type: String
 
+  NodeAutoScalingInitSize:
+      Type: Number
+      Description: The initial size of Node Group ASG.
+      Default: 1
+
+  NodeAutoScalingGroupMinSize:
+      Type: Number
+      Description: Minimum size of Node Group ASG.
+      Default: 1
+
+  NodeAutoScalingGroupMaxBatchSize:
+      Type: Number
+      Description: Minimum size of Node Group ASG.
+      Default: 2
```

Comment:
This is a custom parameter related to the upgrade process, so it does not have to be in the middle of the autoscaling settings. (Custom params could be at the end of the list?)

Reply:
In latest CF template parameter are alphabetically ordered, that's why I ordered the same way to be easily comparable. We may put custom params at the end in same order.

Reply:
reordered params to have additional params at the end